analysis_file/lib/analysis_keys.py

```diff
@@ -25,9 +25,16 @@ def set_matrix_keys(user, list_td, keys_to_matrix):
             matrix_d = dict()
             for key, prefix in keys_to_matrix.items():
                 if key in td:
-                    new_key = td[key] + prefix
-                    matrix_keys.add(new_key)
-                    matrix_d[new_key] = Codes.MATRIX_1
+                    if type(td[key]) is list:
+                        for code in td[key]:
+                            new_key = prefix + code
+                            matrix_keys.add(new_key)
+                            matrix_d[new_key] = Codes.MATRIX_1
+                            td.append_data(matrix_d, Metadata(user, Metadata.get_call_location(), time.time()))
+                    else:
+                        new_key = prefix + td[key]
```

Comment:
might be worth a comment here to say that you're assuming td[key] is a label/dict in this case

Reply:
I've added it